utility/types/error.go

```diff
@@ -325,7 +330,7 @@ func ErrUnequalRounds() Error {
 	return NewError(CodeUnequalRoundsError, UnequalRoundsError)
 }
 
-func ErrInvalidServiceUrl(reason string) Error {
+func ErrInvalidServiceURL(reason string) Error {
```

Comment:
Another example of inconsistency around `URL` / `Url`. I recommend a case-sensitive search and renaming consistently across the board.